pkg/storage/ingest/config.go

```diff
@@ -115,6 +120,9 @@ func (cfg *KafkaConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet)
 	f.DurationVar(&cfg.WriteTimeout, prefix+".write-timeout", 10*time.Second, "How long to wait for an incoming write request to be successfully committed to the Kafka backend.")
 	f.IntVar(&cfg.WriteClients, prefix+".write-clients", 1, "The number of Kafka clients used by producers. When the configured number of clients is greater than 1, partitions are sharded among Kafka clients. A higher number of clients may provide higher write throughput at the cost of additional Metadata requests pressure to Kafka.")
 
+	f.StringVar(&cfg.SASLUsername, prefix+".sasl-username", "", "The username used to authenticate to Kafka using the SASL plain mechanism. To enable SASL, configure both the username and password.")
```

Comment:
nit: I'm not sure which patterns we follow and if its common that when requiring multiple config to be set it basically no-ops unless you set them all. **The flag documentation is very clear** that you need both hence why this is a nit, but...

I can see from the library that:

```
	if auth.User == "" || auth.Pass == "" {
		return nil, nil, errors.New("PLAIN user and pass must be non-empty")
	}
```

hence the reasoning behind not setting it unless you get both, but I wonder if it's convenient to validate that here or let the error bubble up.

I'll leave the decision in your hands and this is considered a very nitpicky nit.

Reply:
I think we can add a validation in our config. Let me do it.

Reply:
Done in https://github.com/grafana/mimir/pull/9584/commits/8cef267d67233fb7f72d99129296fdb8aac1f2db